core/vm/logger_trace.go

```diff
@@ -0,0 +1,126 @@
+package vm
+
+import (
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/common/hexutil"
+	"github.com/scroll-tech/go-ethereum/core/types"
+)
+
+type traceFunc func(l *StructLogger, scope *ScopeContext, extraData *types.ExtraData) error
+
+var (
+	// OpcodeExecs the map to load opcodes' trace funcs.
+	OpcodeExecs = map[OpCode][]traceFunc{
+		CALL:         {traceToAddressCodeHash, traceLastNAddressCodeHash(1), traceOriginProof, traceLastNAddressProof(1)},
+		CALLCODE:     {traceToAddressCodeHash, traceLastNAddressCodeHash(1), traceOriginProof, traceLastNAddressProof(1)},
```

Comment:
using `traceOriginProof` here can be wrong. caller may not be the tx sender.

we can retrieve caller_address using "ScopeContext.Contract.CallerAddress"

Reply:
[This line](https://github.com/scroll-tech/go-ethereum/blob/8b16b4cefae82f7d3f73a37df9e04d7d356a7a23/core/evm.go#L70) can show origin is tx sender address.

This is a mistake we only need contract caller proof, Don't tx's sender proof, that right?

Reply:
Yes, that's what I mean. We need caller instead of sender.  sometime the caller is the sender, but sometimes not